bindings/java/java/src/main/java/org/ethereum/evmc/EvmcVm.java

```diff
@@ -7,37 +7,24 @@
 import static org.ethereum.evmc.Host.removeContext;
 
 import java.nio.ByteBuffer;
-import java.util.Objects;
 
 /**
  * The Java interface to the evm instance.
  *
  * <p>Defines the Java methods capable of accessing the evm implementation.
  */
 public final class EvmcVm implements AutoCloseable {
-  private static EvmcVm evmcVm;
-  private static boolean isEvmcLibraryLoaded = false;
   private ByteBuffer nativeVm;
+
   /**
    * This method loads the specified evm shared library and loads/initializes the jni bindings.
    *
+   * @param evmcPath the path to the evmc shared library
    * @param filename /path/filename of the evm shared object
    */
-  public static EvmcVm create(String filename) {
-    if (!EvmcVm.isEvmcLibraryLoaded) {
-      try {
-        // load so containing the jni bindings to evmc
-        System.load(System.getProperty("user.dir") + "/../c/build/lib/libevmc-java.so");
-        EvmcVm.isEvmcLibraryLoaded = true;
-      } catch (UnsatisfiedLinkError e) {
```

Comment:
Couldn't `System.load` still emit this exception? Shouldn't `create` specify `throws Exception`?

Reply:
System.load does not throw this exception explicitly, so it doesn't have to be caught there. It can be caught by the code creating the EVM to try different paths in succession. The current code would run `System.exit(1)` if the lib was not found, which is a big problem if you're trying different paths.

Note also System.load is idempotent, so there's no need to check if it's loaded or not.

Reply:
It does seem to throw exceptions: https://docs.oracle.com/javase/10/docs/api/java/lang/System.html#load(java.lang.String)

Reply:
The exceptions are documented in the Javadoc but not in the signature of the method.